Expose most_specific_header_mutations_wins (#8692)
solo-io/gloo: docs/content/guides/traffic_management/request_processing/append_remove_headers/_index.md

In the previous example of the `x-route-table` response header, the virtual host level header overwrites the route level header because the virtual host level header is evaluated after the route level header.
If you set `append: true` or omit this field on the virtual host, then the route level response header (`a`) would get appended to the virtual host level header (`alphabet`).
If you set `append: false` on the route, the route does not affect the virtual host because the route is evaluated before the virtual host. In the example, the response header would stay `x-route-table: alphabet`.

### Reversing the order of header manipulation evaluation

You can reverse the order in which header manipulations are evaluated so that order of evaluation becomes: global level headers, virtual host level headers, route level headers, and finally weighted cluster level headers.

> With the order of evaluation being reversed, more specific header manipulations can override less specific ones.
> This can be done by setting the `mostSpecificHeaderMutationsWins` field to `true` in the [routeOptions]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options.proto.sk/#routeconfigurationoptions" >}}) settings for a [Gateway]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/gateway.proto.sk/" >}}).

Reviewer: Break up paragraphs:
```suggestion
With the order of evaluation being reversed, more specific header manipulations can override less specific ones.

To reverse the order of evaluation, set the `mostSpecificHeaderMutationsWins` field to `true` in the [routeOptions]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/options.proto.sk/#routeconfigurationoptions" >}}) settings for a [Gateway]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/gateway.proto.sk/" >}}).
```

The route options that you set on the Gateway will apply to all routes that the Gateway serves.

```yaml
apiVersion: gateway.solo.io/v1
kind: Gateway
metadata: # collapsed for brevity
spec:
  routeOptions:
    mostSpecificHeaderMutationsWins: true
status: # collapsed for brevity
```

> Using our VirtualService and RouteTable setup above, we can set the `append: false` on the RouteTable:

Reviewer: ```suggestion
Following the VirtualService and RouteTable setup in the previous section, add the `append` setting to `false` on the RouteTable. This way, the header values added by the RouteTable override less specific headers instead of append to the values.
```

{{< highlight yaml "hl_lines=25-25" >}}
apiVersion: gateway.solo.io/v1
kind: RouteTable
metadata:
  name: 'a-routes'
  namespace: 'a'
spec:
  routes:
    - matchers:
        # the path matchers in this RouteTable must begin with the prefix `/a/`
      - prefix: '/a/1'
      routeAction:
        single:
          upstream:
            name: 'foo-upstream'
  options:
    headerManipulation:
      requestHeadersToAdd:
        - header:
            key: x-route-table
            value: a
      responseHeadersToAdd:
        - header:
            key: x-route-table
            value: a
          append: false
{{< /highlight >}}

> Now, because we have `mostSpecificHeaderMutationsWins` set to `true`, the `x-route-table` response header in the child overwrites the parents object's value of `alphabet` instead to `a`.

Reviewer: ```suggestion
With `mostSpecificHeaderMutationsWins` set to `true` and `append` set to `false`, now the `x-route-table` response header in the child RouteTable overwrites the parent object's value of `alphabet` instead to `a`.
```
